tools/build-cocoa.sh

```diff
@@ -85,57 +85,63 @@ for bt in "${BUILD_TYPES[@]}"; do
-            mv "core/lib/librealm-parser-${SDK}-simulator${suffix}.a" "core/librealm-parser-${SDK}-simulator${suffix}.a"
-        fi
-        rm -rf core/lib
     done
 done
 
+function extract_slices {
+    local input="$1"
+    local output="$2"
+    local archs="$3"
+
+    # create-xcframework does not like fat libraries with a single arch but
+    # also doesn't like two thin libraries for one platform, so we have to
+    # create different types of libraries depending on the arch count
+    if [ $(wc -w <<< "$archs") == 2 ]; then
+        archs=$(sed 's/extract/thin/' <<< "$archs")
+    fi
+    lipo $archs -output "$output" "$input"
+}
+
 # Produce xcframeworks
-if [[ ! -z $BUILD_XCFRAMEWORK ]]; then
+if [[ -n $BUILD_XCFRAMEWORK ]]; then
     for bt in "${BUILD_TYPES[@]}"; do
         make_core_xcframework=()
         [[ "$bt" = "Release" ]] && suffix="" || suffix="-dbg"
+        rm -rf xcf-tmp
+        mkdir xcf-tmp
         for p in "${PLATFORMS[@]}"; do
             if [[ "$p" == "macosx" ]]; then
                 build_dir="core/librealm-macosx${suffix}.a"
-                make_core_xcframework+=( -library ${build_dir} -headers core/include/)
+                make_core_xcframework+=( -library "${build_dir}")
             elif [[ "$p" == "maccatalyst" ]]; then
                 build_dir="core/librealm-maccatalyst${suffix}.a"
-                make_core_xcframework+=( -library ${build_dir} -headers core/include/)
+                make_core_xcframework+=( -library "${build_dir}")
             else
-                case "${p}" in
-                    ios) SDK="iphone";;
-                    watchos) SDK="watch";;
-                    tvos) SDK="appletv";;
-                esac
-                build_dir_device="core/librealm-${SDK}-device${suffix}.a"
-                build_dir_simulator="core/librealm-${SDK}-simulator${suffix}.a"
-                make_core_xcframework+=( -library ${build_dir_device} -headers core/include/)
-                make_core_xcframework+=( -library ${build_dir_simulator} -headers core/include/)
+                build_dir="core/librealm-${p}${suffix}.a"
+                output_prefix="xcf-tmp/librealm-${p}${suffix}"
+                device=''
+                simulator=''
+                for arch in $(lipo -archs "$build_dir"); do
+                    if [[ "$arch" == arm* ]]; then
+                        device+="-extract $arch "
+                    else
+                        simulator+="-extract $arch "
+                    fi
+                done
+                extract_slices "$build_dir" "${output_prefix}-device.a" "$device"
+                extract_slices "$build_dir" "${output_prefix}-simulator.a" "$simulator"
+                make_core_xcframework+=( -library "${output_prefix}-device.a")
+                make_core_xcframework+=( -library "${output_prefix}-simulator.a")
             fi
         done
+        make_core_xcframework+=( -headers core/include)
```

Comment:
`-headers core/include` needs to be appended to each `-library "${output_prefix}-XXX.a"`. I built the branch locally and only the `tvos-x86_64-simulator` folder within the xcframework contains headers.